logstash-core/src/main/java/org/logstash/Logstash.java

```diff
@@ -215,6 +215,39 @@ public void close() {
         ruby.tearDown(false);
     }
 
+    /**
+     * Initialize a runtime configuration.
+     * @param lsHome the LOGSTASH_HOME
+     * @param args extra arguments (ARGV) to process
+     * @return a runtime configuration instance
+     */
+    public static RubyInstanceConfig initRubyConfig(final Path lsHome,
+                                                    final String... args) {
+        return initRubyConfigImpl(lsHome.toString(), safePath(lsHome, "vendor", "jruby"), args);
+    }
+
+    /**
+     * Initialize a runtime configuration.
+     * @param lsHome the LOGSTASH_HOME
+     * @param args extra arguments (ARGV) to process
+     * @return a runtime configuration instance
+     */
+    public static RubyInstanceConfig initRubyConfig(final Path lsHome,
+                                                    final String currentDir,
+                                                    final String... args) {
+        return initRubyConfigImpl(currentDir, safePath(lsHome, "vendor", "jruby"), args);
+    }
+
+    private static RubyInstanceConfig initRubyConfigImpl(final String currentDir,
+                                                     final String jrubyHome,
+                                                     final String[] args) {
+        final RubyInstanceConfig config = new RubyInstanceConfig();
+        if (currentDir != null) config.setCurrentDirectory(currentDir);
+        config.setJRubyHome(jrubyHome);
+        config.processArguments(args);
+        return config;
+    }
+
```

Comment:
yes it can be smt else, in theory e.g. a `classpath://...` like entry.
however should be fine with a `Path` type (for now) on the public methods - good point, updated!